src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -519,6 +519,20 @@ void DeleteInvalidBlocks(Keccak invalidBlockHash)
 
                     blocksToProcess.Add(block);
                 }
+
+                if (!blocksToProcess[0].IsGenesis)
+                {
+                    BlockHeader? parentOfFirstBlock = _blockTree.FindHeader(blocksToProcess[0].ParentHash!, BlockTreeLookupOptions.None);
+                    if (parentOfFirstBlock == null)
+                    {
+                        throw new InvalidOperationException("Attempted to process a disconnected blockchain");
+                    }
+
+                    if (!_stateReader.HasStateForBlock(parentOfFirstBlock))
+                    {
+                        throw new InvalidOperationException("Attempted to process a blockchain without having starting state");
+                    }
```

Comment:
I expect that some other exception would have been thrown anyway?

Reply:
Well yea, but TrieException. Not exactly clear whats wrong.